habitat/plan.sh

```diff
@@ -64,40 +58,46 @@ do_prepare() {
   build_line "Ruby ABI version appears to be ${RUBY_ABI_VERSION}"
 
   build_line "Setting link for /usr/bin/env to 'coreutils'"
-  [[ ! -f /usr/bin/env ]] && ln -s $(pkg_path_for coreutils)/bin/env /usr/bin/env
-
-  return 0
+  [ ! -f /usr/bin/env ] && ln -s "$(pkg_path_for coreutils)/bin/env" /usr/bin/env || return 0
 }
 
 do_build() {
-  cd $CACHE_PATH
-  export CPPFLAGS="${CPPFLAGS} ${CFLAGS}"
-
-  local _bundler_dir=$(pkg_path_for bundler)
-  local _libxml2_dir=$(pkg_path_for libxml2)
-  local _libxslt_dir=$(pkg_path_for libxslt)
-  local _zlib_dir=$(pkg_path_for zlib)
-
-  export GEM_HOME=${pkg_prefix}
-  export GEM_PATH=${_bundler_dir}:${GEM_HOME}
-
-  export NOKOGIRI_CONFIG="--use-system-libraries --with-zlib-dir=${_zlib_dir} --with-xslt-dir=${_libxslt_dir} --with-xml2-include=${_libxml2_dir}/include/libxml2 --with-xml2-lib=${_libxml2_dir}/lib --without-iconv"
-  bundle config --local build.nokogiri "${NOKOGIRI_CONFIG}"
-
-  bundle config --local silence_root_warning 1
-
-  bundle install --without dep_selector --no-deployment --jobs 2 --retry 5 --path $pkg_prefix
-
-  gem build chef-dk.gemspec
-  gem install --no-doc chef-dk-*.gem
+  local _bundler_dir
+  local _libxml2_dir
+  local _libxslt_dir
+  local _zlib_dir
+  export NOKOGIRI_CONFIG
+  export GEM_HOME
+  export GEM_PATH
```

Comment:
It is safer to first initialize the variables and then set the values, especially if you don't know what the content of the variable could be.

Reply:
That's a pattern that has come from when the value is from a command interpolation, right? Because the exit code gets eaten and testing success would be on `local`? So, just get in the habit of initializing locals first, then assigning values?